tools/ipapy2/ipapy2_map_isotope_patterns.xml

```diff
@@ -0,0 +1,62 @@
+<tool id="ipapy2_map_isotope_patterns" name="IPA map isotope patterns" version="@TOOL_VERSION@+galaxy0" profile="21.05">
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">ipapy2</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '${ipapy2_map_isotope_cli}'
+    ]]></command>
+
+<configfiles>
+<configfile name="ipapy2_map_isotope_cli">
+@init_import@
+
+clustered_df = pd.read_csv('${clustered_MS1_intensities}')
+<!-- The function call below updates the passed dataframe in place hence it returns nothing -->
+ipa.map_isotope_patterns(clustered_df, isoDiff=$isotope.isoDiff, ppm=$isotope.ppm, ionisation=int(${ionisation}), MinIsoRatio=$isotope.isotope_ratio)
+clustered_df.to_csv('${mapped_isotope_patterns}', index=False)
+</configfile>
+</configfiles>
+
+    <inputs>
+        <param label="IPA dataframe" name="clustered_MS1_intensities" type="data" format="csv" help="A csv file of clustered MS1 intensities." />
+        <expand macro="ionisation"/>
+        <section name="isotope" title="isotope settings">
+            <param name="isoDiff" type="float" value="1">
+                <label>isotope difference</label>
+                <help>Difference between isotopes of charge 1.</help>
+            </param>
+            <param name="ppm" type="float" value="100">
```

Comment:
I don't fully understand this help text.

Reply:
I think it was a copy and paste from the documentation:
https://github.com/francescodc87/ipaPy2/blob/bebd9126428bba9e64845e057df4892dd7c88b7c/README.md?plain=1#L1006

I am assuming it means - the difference between isotopes having a charge of 1